### PR TITLE
add weblink to node fields

### PR DESCRIPTION
**Pull request by durandom**
image

Add the source as a field, so that we can link to it from a page, much like a "fork me on GH" ribbon.
Any ideas on a Patternfly component? Or where to put the link? 

cc @cfchase @tumido @oindrillac

### CODE REVIEW

**Comment by oindrillac** [comment]
We can have a Button https://www.patternfly.org/v4/components/button if we are to add something like :-

* A `Fork Me on Github` button on the top left after the Header like the "Try it free" button here https://cloud.google.com/dialogflow#cx-agent
image

* Or a links section containing items like `Fork me on Github`, `Try it on Jupyterhub` (this could route to the anchor link for the sub-section on the page describing the steps) etc similar to the "Links" section on the right here https://airbnb.io/projects/rx-groups/

image

**Comment by tumido** [comment]
I think call to action button would look the best here:
image

https://www.patternfly.org/v4/components/button#call-to-action

**Comment by durandom** [comment]
Not really beautiful, but does the job:

image

File: gatsby-node.js [hunk 1]
```diff
@@ -41,6 +41,22 @@ exports.onCreateNode = ({ node, getNode, actions }) => {
...
       value: slug
     })
+
+    let srcLink = "";
+    if (gitRemoteNode) {
+      srcLink = gitRemoteNode.webLink;
+      srcLink = srcLink + "/blob/master/" + fileNode.relativePath;
+    } else {
+      // FIXME: use something configurable...
+      srcLink = "https://github.com/operate-first/operate-first.github.io";
```
Comment (durandom) [inline gatsby-node.js]: @cfchase @tumido @martinpovolny any idea how to get this URL programatically? 
And even if we just put it as a config option into gatsby-config.js

Comment (durandom) [reply gatsby-node.js]: nvm, I got it:

```
    let siteMetadata = getNode('Site').siteMetadata;
```

File: gatsby-node.js [hunk 2]
```diff
@@ -41,6 +41,22 @@ exports.onCreateNode = ({ node, getNode, actions }) => {
       name: "slug",
       value: slug
     })
+
+    let siteMetadata = getNode('Site').siteMetadata;
```
Comment (tumido) [inline gatsby-node.js]: Do we need this variable? It's referred to it only once in the else branch. What if we use it directly?

File: gatsby-node.js [hunk 3]
```diff
@@ -41,6 +41,22 @@ exports.onCreateNode = ({ node, getNode, actions }) => {
...
+    let srcLink = "";
+    if (gitRemoteNode) {
+      srcLink = gitRemoteNode.webLink;
+      srcLink = srcLink + "/blob/master/" + fileNode.relativePath;
+    } else {
+      srcLink = siteMetadata.srcLinkDefault;
+      srcLink = srcLink + "/blob/master/content/" + fileNode.relativePath;
+    }
```
Comment (tumido) [inline gatsby-node.js]: can this be `const` instead? https://eslint.org/docs/rules/prefer-const

```suggestion
    const srcBase = gitRemoteNode ? `${gitRemoteNode.webLink}/blob/master/` : `${siteMetadata.srcLinkDefault}/blob/master/content/`
```

and later on use it as

```jsx
    createNodeField({
      node,
      name: "srcLink",
      value: srcBase + fileNode.relativePath
    })
```

File: src/components/Layout/Layout.js [hunk 4]
```diff
@@ -20,6 +20,13 @@ export const Layout = ({ location, title, children }) => {
       className="layout"
     >
       {children}
+      <PageSection variant={PageSectionVariants.dark}>
+        <TextContent>
+          <Button variant="primary" isLarge component="a" href={srcLink} target="_contribute">
+            Contribute to this page at {srcLink}
```
Comment (tumido) [inline src/components/Layout/Layout.js]: Do we need to show the link in the button text though? They can get pretty long and "Contribute to this page" call to action seems obvious enough to me.

**Review by tumido — COMMENTED** [review]
We can refine the looks of it later if we don't like it, I think it's fine for now. I have few nitpicks code suggestions and what not, see above. Otherwise LGTM :+1:

File: src/components/Layout/Layout.js (hunk 4 above)
Comment (durandom) [reply src/components/Layout/Layout.js]: I was undecided. Now we have a vote. I'll change

**Comment by durandom** [comment]
@tumido thanks for the JS style. My coding skills are indeed a bit rusty and can use some polish, appreciated.
How about now?

**Review by tumido — APPROVED** [review]
LGTM :+1:

**Pull request merged**